scene/gui/text_edit.cpp

```diff
@@ -3445,6 +3445,7 @@ void TextEdit::set_line(int p_line, const String &p_new_text) {
 
 		if (has_selection(i) && p_line == get_selection_to_line(i) && get_selection_to_column(i) > text[p_line].length()) {
 			carets.write[i].selection.to_column = text[p_line].length();
+			carets.write[i].selection.selecting_column = CLAMP(carets[i].selection.selecting_column, 0, text[p_line].length());
```

Comment:
Would add a new check for `selecting_column` as it's not guaranteed that `get_selection_to_line() == selecting_line`